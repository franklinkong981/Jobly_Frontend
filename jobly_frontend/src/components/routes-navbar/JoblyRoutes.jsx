import React from "react";
import {Routes, Route, Navigate} from "react-router-dom";

import Homepage from "../home/Homepage.jsx";
import LoginForm from "../auth/LoginForm.jsx";
import SignupForm from "../auth/SignupForm.jsx";
import UpdateProfileForm from "../profile/UpdateProfileForm.jsx";
import CompanyList from "../companies/CompanyList.jsx";
import CompanyDetail from "../companies/CompanyDetail.jsx";
import JobList from "../jobs/JobList.jsx";

function JoblyRoutes({signUpFunc, loginFunc}) {
  return (
    <div className="Routes pt-5">
      <Routes>
        <Route exact path="/login" element={<LoginForm loginFunc={loginFunc} />} />
        <Route path="/signup" element={<SignupForm signUpFunc={signUpFunc} />} />
        
        <Route path="/profile" element={<UpdateProfileForm />} />
        <Route path="/companies" element={<CompanyList />} />
        <Route path="/companies/:name" element={<CompanyDetail />} />
        <Route path="/jobs" element={<JobList />} />
        <Route exact path="/" element={<Homepage />} />
        <Route path="*" element={<Navigate to="/"/>}/>
      </Routes>
    </div>
  );
}

export default JoblyRoutes;