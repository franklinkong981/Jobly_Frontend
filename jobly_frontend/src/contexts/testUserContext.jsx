import React from "react";
import CurrentUserContext from "./currentUserContext.jsx";

const testUser = {
  username: "testuser",
  first_name: "Testy",
  last_name: "Usery",
  email: "testuser981@gmail.com"
};

const TestUserProvider = ({children, currentUserInfo = testUser, hasUserAppliedToJob= () => false}) => (
  <CurrentUserContext.Provider value={{currentUserInfo, hasUserAppliedToJob}}>
    {children}
  </CurrentUserContext.Provider>
);

export {TestUserProvider};