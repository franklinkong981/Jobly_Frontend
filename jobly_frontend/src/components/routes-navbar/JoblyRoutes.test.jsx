import React from "react";
import {render} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";

import {TestUserProvider} from "../../contexts/testUserContext.jsx";

import JoblyRoutes from "./JoblyRoutes.jsx";
import { expect, it } from "vitest";

it("renders without crashing", function() {
  render (
    <MemoryRouter>
      <TestUserProvider>
        <JoblyRoutes />
      </TestUserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function() {
  const {asFragment} = render (
    <MemoryRouter>
      <TestUserProvider>
        <JoblyRoutes />
      </TestUserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
