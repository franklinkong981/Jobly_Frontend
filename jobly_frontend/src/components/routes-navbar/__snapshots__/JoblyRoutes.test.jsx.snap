// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Routes pt-5"
  >
    <div
      class="Homepage"
    >
      <div
        class="Homepage-logged-in container text-center"
      >
        <h1
          class="Homepage-logged-in-title mb-4 font-weight-bold"
        >
          Welcome back !
        </h1>
        <p
          class="Homepage-logged-in-subtitle lead"
        >
          See companies and job openings below.
        </p>
        <div
          class="Homepage-logged-in-links"
        >
          <a
            class="Homepage-logged-in-companies-link btn btn-secondary font-weight-bold mr-3"
            data-discover="true"
            href="/companies"
          >
            See All Companies
          </a>
          <a
            class="Homepage-logged-in-jobs-link btn btn-info font-weight-bold"
            data-discover="true"
            href="/jobs"
          >
            See Job Openings
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
